threeML/plugins/OGIP/pha.py

```diff
@@ -69,6 +69,8 @@ def write(self, outfile_name, overwrite=True):
         :return:
         """
 
+        self._force_rsp_write = force_rsp_write
```

Comment:
It would be better to simply pass force_rsp_write as a parameter to self._append_ogip.

In general, no attribute of the class should be set for the first time outside the constructor (or if really necessary in methods called by the constructor).

The reason is that sometimes you need to know the state of the class (for example to serialize it with pickle or dill for parallel processing), and a state that changes depending on the interaction with the user is going to be very difficult to handle.